src/app/features/library/search-form/search-form.component.ts

```diff
@@ -0,0 +1,118 @@
+import { HttpErrorResponse } from '@angular/common/http';
+import { of } from 'rxjs';
+
+@Component({
+  selector: 'app-search-form',
+  templateUrl: './search-form.component.html',
+  styleUrls: ['./search-form.component.scss'],
+  providers: [
+    { provide: YT_VIDEO_TYPE, useValue: YT_VIDEO_TYPE },
+    { provide: VIMEO_VIDEO_TYPE, useValue: VIMEO_VIDEO_TYPE },
+  ],
+})
+export class SearchFormComponent implements OnInit {
+
+  public form: FormGroup;
+  private readonly VIDEO_NOT_FOUND_MESSAGE = 'Video not found';
+
+  constructor(
+    private fb: FormBuilder,
+    private helperService: HelperService,
+    private ytDataService: YtDataService,
+    private vimeoDataService: VimeoDataService,
+    private videosStoreService: VideosStoreService,
+    private snackbarService: SnackbarService,
+    @Inject(YT_VIDEO_TYPE) public ytVideoType: string,
+    @Inject(VIMEO_VIDEO_TYPE) public vimeoVideoType: string,
+  ) {}
+
+  public ngOnInit(): void {
+    this.form = this.createForm();
+  }
+
+  private clearInput(): void {
+    this.form.controls.id.reset();
+  }
+
+  public getDemoVideos(): void {
+    this.videosStoreService.getDemoVideos();
+  }
+
+  public onSubmit(): void {
+    const id = this.helperService.extractId(this.form.value.id);
+    const type = this.form.value.type;
+    let foundVideo: SavedVideoData;
+
+    if (id.trim() === '') {
+      return;
+    }
+
+    if (type === YT_VIDEO_TYPE) {
+      this.ytDataService.getVideoByIds([id])
+        .subscribe((response: YtVideosResponse) => {
+          const video = response.items[0];
+
+          if (!video) {
+            this.snackbarService.openErrorSnackbar(this.VIDEO_NOT_FOUND_MESSAGE);
+          } else {
+            foundVideo = {
+              type: YT_VIDEO_TYPE,
+              id: video.id,
+              addedToLibraryAt: new Date(),
+              isFavourite: false,
+            };
+
+            this.videosStoreService.saveNewVideo(foundVideo);
+          }
+        });
+    } else {
+      this.vimeoDataService.getVideoByIds([id])
+        .pipe(
+          catchError((err: HttpErrorResponse) => {
+            if (err.statusText === 'Bad Request') {
+              this.snackbarService.openErrorSnackbar(this.VIDEO_NOT_FOUND_MESSAGE);
+            }
+            return of({});
+          })
+        )
+        .subscribe((response: VimeoResponse) => {
+          if (!response.data[0]) {
+            this.snackbarService.openErrorSnackbar(this.VIDEO_NOT_FOUND_MESSAGE);
+          } else {
+            const video = response.data[0];
+
+            foundVideo = {
+              type: VIMEO_VIDEO_TYPE,
+              id: this.helperService.extractId(video.link),
+              addedToLibraryAt: new Date(),
+              isFavourite: false,
+            };
+
+            this.videosStoreService.saveNewVideo(foundVideo);
+          }
+        });
+    }
+
+    this.clearInput();
+  }
+
+  private createForm(): FormGroup {
+    return this.fb.group({
```

Comment:
Nie uważasz, że dla jednego jedynego malutkiego inputa reactive form, a nawet template driven form, to trochę overkill? Tym bardziej, że nie masz walidacji.